scripts/checks.sh

```diff
@@ -46,7 +49,105 @@ function log_status()
 	jq -cn --argjson g "${1}" --arg f "${2}" --arg s "${3}" '[{"name":$f,"success":$g,"status":$s}]'
 }
 
+function test_upstream_dns()
+{
+	# force dnsmasq to print statistics to journal to read them back
+	systemctl kill -s USR1 dnsmasq
+	for i in $(journalctl -u dnsmasq | awk 'match($0, /[12]?[0-9]?[0-9]\.[12]?[0-9]?[0-9]\.[12]?[0-9]?[0-9]\.[12]?[0-9]?[0-9]/) {print substr($0, RSTART, RLENGTH)}'| sort -u); do
+		# shellcheck disable=SC2001
+		for j in "${external_fqdn}" "$(echo "${API_ENDPOINT}" | sed -e 's@^https*://@@')"; do
+			if ! nslookup "${j}" "${i}" > /dev/null 2>&1; then
+				echo "${FUNCNAME[0]}: DNS lookup failed for ${j} via upstream: ${i}"
+			fi
+		done;
+	done
+}
+
+function test_wifi()
+{
+	local -i wifi_active_interfaces
+	wifi_active_interfaces=$(nmcli --fields TYPE,STATE device status | grep -c '^wifi *connected')
+	if (( wifi_active_interfaces > 0 )); then
+		local -i wifi_strength
+		wifi_strength=$(nmcli -f IN-USE,SIGNAL device wifi list | awk '/^\*/{print $2}')
+		if (( wifi_strength < wifi_threshold )); then
+			echo "${FUNCNAME[0]}: Configured wifi network has a weak signal (<${wifi_threshold}%)"
+		fi
+	fi
+}
+
+function test_ping()
```

Comment:
Should we also measure connection latency here too?

Reply:
+1 for latency, be nice to get the stddev too imo.

Reply:
Packet loss is an easy one (mostly binary condition). @shaunmulligan or @wrboyce do you have an idea of acceptable/unacceptable latencies (since these are checks, not just data collection mechanisms)?

I'm not super comfortable using the stddev here (latencies are notoriously non-normal), but happy to think about other statistics that may give better insight! Perhaps max/min? TIL that `ping`'s `mdev` is actually the stddev... disappointing